From: =?utf-8?q?J=C3=B6rg_Thalheim?= <joerg@higgsboson.tk>
Date: Mon, 19 Dec 2016 15:34:07 +0100
Subject: networkd-ndisc: handle missing mtu gracefully (#4913)

At least bird's implementation of router advertisement does not
set MTU option by default (instead it supplies an option to the user).
In this case just leave MTU as it is.

(cherry picked from commit 29b5ad083a6925efec8e188013d1298742e0baaa)
---
 src/network/networkd-ndisc.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/network/networkd-ndisc.c b/src/network/networkd-ndisc.c
index 9cfdf01..d3fa56b 100644
--- a/src/network/networkd-ndisc.c
+++ b/src/network/networkd-ndisc.c
@@ -117,7 +117,9 @@ static void ndisc_router_process_default(Link *link, sd_ndisc_router *rt) {
         }
 
         r = sd_ndisc_router_get_mtu(rt, &mtu);
-        if (r < 0) {
+        if (r == -ENODATA)
+                mtu = 0;
+        else if (r < 0) {
                 log_link_warning_errno(link, r, "Failed to get default router MTU from RA: %m");
                 return;
         }
